### PR TITLE
HAAR-1033: ⬆️ update to spring boot 3.0.0

### PR DESCRIPTION
**Pull request by petergphillips**
- HAAR-1033: ⬆️ update to spring boot 3.0.0
- HAAR-1033: ⬆️ update java version used in circleci
- HAAR-1033: ⬆️ update java version used in circleci
- HAAR-1033: ⬆️ update java version used in circleci
- SDI-477: ⬆ Bump versions and remove pinning
- SDI-477: ⬆ Change to a major version upgrade

### CODE REVIEW

**Review by andymarke — APPROVED** [review]
LGTM

**Comment by petergphillips** [comment]
Closing in preference to having a `spring-boot-3.0.0` branch

**Comment by petergphillips** [comment]
actually https://github.com/ministryofjustice/dps-gradle-spring-boot/pull/226

File: README.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ This Gradle plugin is used to orchestrate DPS Spring Boot projects such that:
 * CVEs causing `dependencyCheckAnalyze` failures are mitigated in a single place rather than in each and every project
 
 ## Release Notes
+##### [5.0.0](release-notes/5.0.0.md)
+##### [4.8.0](release-notes/4.8.0.md)
```
Comment (mikehalmamoj) [inline README.md:13]: What's on 4.8.0?

Comment (petergphillips) [reply README.md]: good spot, we were going to go to 4.8 but changed our minds and went to 5.

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**